client/proposer/service_test.go

```diff
@@ -119,35 +145,20 @@ func TestIncompleteCollation(t *testing.T) {
 		<-fakeProposer.msgChan
 	}
 
-	want := "Starting proposer service"
-
-	msg := hook.LastEntry()
-	if msg.Message != want {
-		t.Errorf("Incorrect log, wanted %v but got %v", want, msg.Message)
+	if err := waitForLogMsg(hook, "Starting proposer service"); err != nil {
```

Comment:
The problem with this is that this test tell us nothing, the other test made sure that the last log Entry was only `Starting Proposer Service`, but with this it just checks if the `Starting Proposer Service` is there in the logs which it most definitely would be. A better way would be to check for `Starting Proposer Service` and `Stopping Proposer Service` and check that their indexes are only differ by 1, if any more than the test fails

Reply:
Makes total sense, I've modified `waitForLogMsg` to return the index where the message was found, now this tests checks that the starting proposer message is the last one (checking if there's also a stop proposer log message causes an index out of bounds panic, the starting log is indeed the last entry), let me know WDYT.

Reply:
Unfortunately tests are failing =( . The issue is with discovery with the multicast DNS server so sometimes peers cant be found. Maybe another way you can do it is to check that there is no `CollationCreated` log created before the `Stop Proposer` log. So basically you can wait for 'Stop Proposer` log and then just check all the previous indexes to be sure that there is no `CollationCreated` log